comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -241,12 +251,27 @@ func (r *secretResolver) startRefreshRoutine() {
 	if r.ticker != nil || r.refreshInterval == 0 {
 		return
 	}
-	r.ticker = time.NewTicker(r.refreshInterval)
+
+	if r.refreshIntervalScatter {
+		r.scatterDuration = time.Duration(rand.Int63n(int64(r.refreshInterval)))
+		log.Infof("scatterDuration is %s", r.scatterDuration)
+		r.ticker = r.clk.Ticker(r.scatterDuration)
+	} else {
+		r.ticker = r.clk.Ticker(r.refreshInterval)
+	}
```

Comment:
Related to my comment on line 764, let's make sure `scatterDuration` is always initialized.

```suggestion
	if r.refreshIntervalScatter {
		r.scatterDuration = time.Duration(rand.Int63n(int64(r.refreshInterval)))
		log.Infof("first secret refresh will happen in %s", r.scatterDuration)
	} else {
		r.scatterDuration = r.refreshInterval
	}
	r.ticker = r.clk.Ticker(t.r.scatterDuration)
```